.github/workflows/checks.yml

```diff
@@ -0,0 +1,70 @@
+name: CI
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+
+permissions:
+  # Permission for checking out code
+  contents: read
+
+jobs:
+  build:
+    name: Check & Build Provider
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - run: chmod +x ./scripts/gofmtcheck.sh
+      - name: gofmt Check
+        run: ./scripts/gofmtcheck.sh
+      # - name: Generate provider code
+      #   run: go generate
+      # - name: Check generated code for diffs
+      #   run: git diff --exit-code
+      - name: Build
+        run: go build -v
+
+  acceptance:
+    name: Acceptance Tests
+    needs: build
+    runs-on: ubuntu-latest
+    env:
+      ACI_USERNAME: 'ansible_github_ci'
+      ACI_PASSWORD: 'sJ94G92#8dq2hx*K4qh'
+    concurrency:
+        group: tf-aci-ci-test-${{ matrix.apic_host.name }}
+        cancel-in-progress: false
+    strategy:
+      fail-fast: false
+      matrix:
+        apic_host:
+          - name: v4.2
+            url: 'https://173.36.219.68/'
+          - name: v5.2
+            url: 'https://173.36.219.69/'
+          - name: v6.0
+            url: 'https://173.36.219.70/'
+          - name: aws_cloud
+            url: 'https://52.52.20.121/'
+          - name: azure_cloud
+            url: 'https://20.245.236.136/'
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: '1.7.*'
+          terraform_wrapper: false
+      - name: Terraform Acceptance Test (APIC ${{ matrix.apic_host.name }})
+        run: go test github.com/CiscoDevNet/terraform-provider-aci/v2/internal/provider -v -cover -timeout 120m

```

Comment:
Is the timeout sufficient or should we increase to higher number?

Reply:
For now, a happy test is running between 10-15 mins. Hence, I believe 2 hrs should be enough but we can always adjust if it's required.

Reply:
Changed to 5h as discussed.